packages/auth/ios/RNFBAuth/RNFBAuthModule.m

```diff
@@ -939,6 +941,181 @@ - (void)invalidate {
       [[FIRAuth authWithApp:firebaseApp] useEmulatorWithHost: host port:port];
 }
 
+RCT_EXPORT_METHOD(multiFactorEnrollWithPhone:
+  (FIRApp *) firebaseApp
+    :(NSString *) phoneNumber
+    :(RCTPromiseResolveBlock) resolve
+    :(RCTPromiseRejectBlock) reject
+) {
+    FIRUser *user = [FIRAuth authWithApp:firebaseApp].currentUser;
+
+    if (user) {
+        [user.multiFactor getSessionWithCompletion:^(FIRMultiFactorSession * _Nullable session, NSError * _Nullable error) {
+            if (error) {
+                [self promiseRejectAuthException:reject error:error];
+            } else {
```

Comment:
could replicate the "fast return" idea here as well on error conditions. Check for error, reject and return immediately, un-nest the remaining logic so all the un-happy paths are out of the way and we can focus on the happy path and making sure it makes sense